lib/ofp-util.c

```diff
@@ -1811,6 +1811,131 @@ ofputil_encode_load_filter_prog(enum ofp_version ofp_version,
     return request;
 }
 
+enum ofperr
+ofputil_decode_update_map(struct ol_update_map *msg, void **key, void **value, const struct ofp_header *oh)
+{
+    enum ofperr error = 0;
+
+    struct ofpbuf b = ofpbuf_const_initializer(oh, ntohs(oh->length));
+    enum ofpraw raw = ofpraw_pull_assert(&b);
+    if (raw != OFPRAW_NXT_UPDATE_MAP) {
+        return OFPERR_OFPBMC_BAD_TYPE;
+    }
+
+    struct ol_update_map *buffer = ofpbuf_pull(&b, sizeof(struct ol_update_map));
+
+    msg->filter_prog = ntohs(buffer->filter_prog);
+    msg->map_id = ntohs(buffer->map_id);
+    msg->key_size = ntohl(buffer->key_size);
+    msg->value_size = ntohl(buffer->value_size);
+
+    *key = ofpbuf_try_pull(&b, msg->key_size);
+    if (!*key) {
+        VLOG_WARN_RL(&bad_ofmsg_rl, "size of provided map key is incorrect"
+                                    " (%"PRIu32")", msg->key_size);
+        return OFPERR_OFPBMC_BAD_LEN;
+    }
+
+    *value = ofpbuf_try_pull(&b, msg->value_size);
+    if (!*key) {
+        VLOG_WARN_RL(&bad_ofmsg_rl, "size of provided map value is incorrect"
+                                    " (%"PRIu32")", msg->value_size);
+        return OFPERR_OFPBMC_BAD_LEN;
+    }
+
+    return error;
+}
+
+void
+ofputil_get_map_key_value_size(int argc, char **args, ovs_be16 *key_size, ovs_be16 *value_size)
+{
+
+    int k = 0;
+    int value_pos = 0;
+
+    // the "key" need to be provided as a first element
+    if(strncmp(args[k], "key", sizeof("key")) != 0)
+    {
```

Comment:
Open vSwitch's C code only has a newline before `{` when they start the body of functions, not for `if`s, `while`s, etc.